internal/characters/lauma/burst.go

```diff
@@ -0,0 +1,167 @@
+package lauma
+
+import (
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attacks"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/info"
+	"github.com/genshinsim/gcsim/pkg/enemy"
+)
+
+var burstFrames []int
+
+type paleHymnStack struct {
+	endFrame int
+	isC6     bool
+}
+
+func init() {
+	burstFrames = frames.InitAbilSlice(115) // Q -> walk
+	burstFrames[action.ActionAttack] = 111
+	burstFrames[action.ActionCharge] = 110
+	burstFrames[action.ActionSkill] = 110
+	burstFrames[action.ActionDash] = 111
+	burstFrames[action.ActionWalk] = 111
+	burstFrames[action.ActionSwap] = 109
+}
+
+const (
+	burstKey          = "lauma-burst"
+	burstDeerFrame    = 115
+	paleHymnGainFrame = 96
+)
+
+func (c *char) Burst(p map[string]int) (action.Info, error) {
+	c.AddStatus(c1Key, 20*60, true)
+
+	c.Core.Tasks.Add(func() {
+		c.addPaleHymn(18, false)
+		c.AddStatus(burstKey, 15*60, true) // should this be here?
+	}, paleHymnGainFrame)
+
+	c.ConsumeEnergy(8)
+	c.SetCD(action.ActionBurst, 15*60)
+	return action.Info{
+		Frames: func(next action.Action) int {
+			if c.deerStateReady && next == action.ActionCharge {
+				return burstDeerFrame
+			}
+			return burstFrames[next]
+		},
+		AnimationLength: burstFrames[action.InvalidAction],
+		CanQueueAfter:   burstFrames[action.ActionSwap], // earliest cancel
+		State:           action.BurstState,
+	}, nil
+}
+
+func (c *char) setupPaleHymnBuff() {
+	c.Core.Events.Subscribe(event.OnEnemyHit, func(args ...any) bool {
+		_, ok := args[0].(*enemy.Enemy)
+		if !ok {
+			return false
+		}
+		if len(c.paleHymnStacks) == 0 {
+			return false
+		}
+
+		ae := args[1].(*info.AttackEvent)
+		em := c.Stat(attributes.EM)
+
+		switch ae.Info.AttackTag {
+		case attacks.AttackTagBurningDamage | attacks.AttackTagBloom | attacks.AttackTagHyperbloom | attacks.AttackTagBurgeon:
+			ae.Info.FlatDmg += em * bloomDmgIncrease[c.TalentLvlBurst()]
+
+			if c.Base.Cons >= 2 {
```

Comment:
Move the c2 bonus damage to cons.go

```
func (c *char) c2PaleHymnScaling(isLunar bool) float64 {
    if c.Base.Cons < 2 {
         return 0
    }
    if isLunar {
        return 4
    }
    return 5
}
```

Then here you can just call `ae.Info.FlatDmg += em * c.c2PaleHymnScaling(false)`